src/sidebar/services/frame-sync.ts

```diff
@@ -130,6 +130,17 @@ export class FrameSyncService {
   private _portFinder: PortFinder;
   private _store: SidebarStore;
 
+  /**
+   * ID of an annotation that should be scrolled to after anchoring completes.
+   *
+   * This is set when {@link scrollToAnnotation} is called and the document
+   * needs to be navigated to a different URL. This can happen in EPUBs.
+   *
+   * We store an ID rather than a tag because the navigation currently triggers
+   * a reload of annotations, which will change their tags but not their IDs.
+   */
+  private _pendingScrollToId: string | null;
```

Comment:
In future, once the problem of the sidebar reloading annotations after a navigation is solved, it should be possible to change this field to store an annotation tag rather than an ID.